servicex/code_generator_service/ast_translator.py

```diff
@@ -48,15 +49,22 @@ def __init__(self, message: str):
 
 
 class AstTranslator:
-    def __init__(self, xaod_executor: Optional[atlas_xaod_executor] = None):
+    def __init__(self, executor: Optional[Union[executor, str]] = None):
         '''
         Create the ast translator objects
 
         Arguments
 
-            xaod_executor       The object that will do the translation
+            executor    The object that will do the translation if an executor is specified or
+                        if its a string then appropriate executor is chosen
         '''
-        self._exe = xaod_executor if xaod_executor is not None else atlas_xaod_executor()
+        if isinstance(executor, str) or executor is None:
+            if executor == 'CMS AOD':
+                self._exe = cms_aod_executor()
+            else:
+                self._exe = atlas_xaod_executor()
```

Comment:
Any reason not to catch the bad string here? So that if someone says "foo bar" we crash?